src/Server/RestRouterHandlers/SearchHandler.cpp

```diff
@@ -176,4 +178,66 @@ std::shared_ptr<WriteBufferFromHTTPServerResponse> SearchHandler::getOutputBuffe
 
     return out;
 }
+
+void SearchHandler::setQuerySettings() const
+{
+    static const NameSet reserved_param_names{
+        "compress",
+        "decompress",
+        "user",
+        "password",
+        "quota_key",
+        "query_id",
+        "stacktrace",
+        "buffer_size",
+        "wait_end_of_query",
+        "session_id",
+        "session_timeout",
+        "session_check"};
+
+    auto param_could_be_skipped = [&](const String & name) {
+        /// Empty parameter appears when URL like ?&a=b or a=b&&c=d. Just skip them for user's convenience.
+        if (name.empty())
+            return true;
+
+        if (reserved_param_names.count(name))
+            return true;
+
+        return false;
+    };
+
+    const auto & query_id = getQueryParameter("x-bdg-request-id");
```

Comment:
Didn't we already setup query Id at the very beginning of HTTPHandler ?

Reply:
In RestHTTPRequestHandler.cpp, it add query_id by "X-ClickHouse-Query-Id" which is the same as Clickhouse native http handler. However, here it overrides the query_id if a new header presents, which is the behavior of frontier service. In my opinion, the new search api will keep the same behavior as frontier, so I keep it.

Reply:
In `RestHTTPRequestHandler.cpp` we already have `request_context->setCurrentQueryId(params.get("query_id", request.get("X-ClickHouse-Query-Id", "")));`, i think it's better to do query Id setup there. And we have idempotent-id setup there as well like 

```
    String idem_key = request.get("X-ClickHouse-Idempotent-Id", "");
    if (idem_key.empty())
    {
        idem_key = request.get("x-bdg-idempotent-id", "");
    }
```